libr/io/io_cache.c

```diff
@@ -169,6 +169,9 @@ R_API bool r_io_cache_write(RIO *io, ut64 addr, const ut8 *buf, int len) {
 	}
 	memcpy (ch->data, buf, len);
 	r_list_append (io->cache, ch);
+	if (io->on_write) {
```

Comment:
I think in theory an `REvent` would be better for these kind of things, as it allows more than one callback to be called when an event occurs.